internal/test_cases/receive_value_test_case.go

```diff
@@ -22,10 +22,12 @@ func (t *ReceiveValueTestCase) Run(client *resp_client.RespConnection, logger *l
 	if err != nil {
 		return err
 	}
-
 	t.ActualValue = value
+	return t.AssertAndCheckUnread(client, logger)
+}
 
-	if err = t.Assertion.Run(value); err != nil {
+func (t *ReceiveValueTestCase) AssertAndCheckUnread(client *resp_client.RespConnection, logger *logger.Logger) error {
```

Comment:
```suggestion
func (t *ReceiveValueTestCase) Assert(client *resp_client.RespConnection, logger *logger.Logger) error {
```

(We might not actually be checking for unread here if `ShouldSkipUnreadDataCheck` is set)